crates/modules/process-monitor/probes.bpf.c

```diff
@@ -96,19 +124,76 @@ struct pending_dead_process {
 // Temporary map used to communicate the list of orphaned processes from
 // `sched_process_exit` to `sched_switch`. There we'll be able to read
 // the new parent of the orphans and emit an EVENT_CHANGE_PARENT.
-struct {
+struct
+{
   __uint(type, BPF_MAP_TYPE_QUEUE);
   __type(value, struct pending_dead_process);
   __uint(max_entries, MAX_PENDING_DEAD_PARENTS);
 } orphans_map SEC(".maps");
 
-struct {
+struct
+{
   __uint(type, BPF_MAP_TYPE_HASH);
   __type(key, u64);
   __type(value, u8);
   __uint(max_entries, MAX_CONTAINER_RUNTIMES);
 } container_runtimes_map SEC(".maps");
 
+static __always_inline int get_cgroup_name(char *buf, size_t sz, int *offset, struct task_struct *cur_tsk)
+{
+
+  // int cgrp_id = memory_cgrp_id;
+  // int cgrp_id = cpu_cgrp_id;
+  int cgrp_id = 4;
```

Comment:
From my understanding, all existing cgroup types should have the container information. Have you tried using `0` here?

Reply:
I'm working on it

Reply:
It's not possible to use a random `cgrp_id`.

This is the dump of my `vmlinux`:

```c
enum cgroup_subsys_id {
	cpuset_cgrp_id = 0,
	cpu_cgrp_id = 1,
	cpuacct_cgrp_id = 2,
	io_cgrp_id = 3,
	memory_cgrp_id = 4,
	devices_cgrp_id = 5,
	freezer_cgrp_id = 6,
	net_cls_cgrp_id = 7,
	perf_event_cgrp_id = 8,
	net_prio_cgrp_id = 9,
	hugetlb_cgrp_id = 10,
	pids_cgrp_id = 11,
	rdma_cgrp_id = 12,
	misc_cgrp_id = 13,
	CGROUP_SUBSYS_COUNT = 14,
};
```

and if I print the value for every subsys I get:

- Docker
```console
bpf_trace_printk: tgid: 20713 | cgroup_id: 0 cgroup_id_name: docker-95fec965a5ca8fa0ed8b8e708d95a306e36cbcbb69deac128c2e59aa85eedeb5.scope
bpf_trace_printk: tgid: 20713 | cgroup_id: 1 cgroup_id_name: docker-95fec965a5ca8fa0ed8b8e708d95a306e36cbcbb69deac128c2e59aa85eedeb5.scope
bpf_trace_printk: tgid: 20713 | cgroup_id: 2 cgroup_id_name: 
bpf_trace_printk: tgid: 20713 | cgroup_id: 3 cgroup_id_name: docker-95fec965a5ca8fa0ed8b8e708d95a306e36cbcbb69deac128c2e59aa85eedeb5.scope
bpf_trace_printk: tgid: 20713 | cgroup_id: 4 cgroup_id_name: docker-95fec965a5ca8fa0ed8b8e708d95a306e36cbcbb69deac128c2e59aa85eedeb5.scope
bpf_trace_printk: tgid: 20713 | cgroup_id: 5 cgroup_id_name: 
bpf_trace_printk: tgid: 20713 | cgroup_id: 6 cgroup_id_name: 
bpf_trace_printk: tgid: 20713 | cgroup_id: 7 cgroup_id_name: 
bpf_trace_printk: tgid: 20713 | cgroup_id: 8 cgroup_id_name: docker-95fec965a5ca8fa0ed8b8e708d95a306e36cbcbb69deac128c2e59aa85eedeb5.scope
bpf_trace_printk: tgid: 20713 | cgroup_id: 9 cgroup_id_name: 
bpf_trace_printk: tgid: 20713 | cgroup_id: 10 cgroup_id_name: docker-95fec965a5ca8fa0ed8b8e708d95a306e36cbcbb69deac128c2e59aa85eedeb5.scope
bpf_trace_printk: tgid: 20713 | cgroup_id: 11 cgroup_id_name: docker-95fec965a5ca8fa0ed8b8e708d95a306e36cbcbb69deac128c2e59aa85eedeb5.scope
bpf_trace_printk: tgid: 20713 | cgroup_id: 12 cgroup_id_name: docker-95fec965a5ca8fa0ed8b8e708d95a306e36cbcbb69deac128c2e59aa85eedeb5.scope
bpf_trace_printk: tgid: 20713 | cgroup_id: 13 cgroup_id_name: docker-95fec965a5ca8fa0ed8b8e708d95a306e36cbcbb69deac128c2e59aa85eedeb5.scope
bpf_trace_printk: tgid: 20713 | cgroup_id: 14 cgroup_id_name:
```

- Podman rootless
```console
bpf_trace_printk: tgid: 21066 | cgroup_id: 0 cgroup_id_name: user.slice
bpf_trace_printk: tgid: 21066 | cgroup_id: 1 cgroup_id_name: container
bpf_trace_printk: tgid: 21066 | cgroup_id: 2 cgroup_id_name: 
bpf_trace_printk: tgid: 21066 | cgroup_id: 3 cgroup_id_name: user.slice
bpf_trace_printk: tgid: 21066 | cgroup_id: 4 cgroup_id_name: container
bpf_trace_printk: tgid: 21066 | cgroup_id: 5 cgroup_id_name: 
bpf_trace_printk: tgid: 21066 | cgroup_id: 6 cgroup_id_name: 
bpf_trace_printk: tgid: 21066 | cgroup_id: 7 cgroup_id_name: 
bpf_trace_printk: tgid: 21066 | cgroup_id: 8 cgroup_id_name: container
bpf_trace_printk: tgid: 21066 | cgroup_id: 9 cgroup_id_name: 
bpf_trace_printk: tgid: 21066 | cgroup_id: 10 cgroup_id_name: user.slice
bpf_trace_printk: tgid: 21066 | cgroup_id: 11 cgroup_id_name: container
bpf_trace_printk: tgid: 21066 | cgroup_id: 12 cgroup_id_name: user.slice
bpf_trace_printk: tgid: 21066 | cgroup_id: 13 cgroup_id_name: user.slice
bpf_trace_printk: tgid: 21066 | cgroup_id: 14 cgroup_id_name:
```

- Podman rootful
```console
bpf_trace_printk: tgid: 22345 | cgroup_id: 0 cgroup_id_name: container
bpf_trace_printk: tgid: 22345 | cgroup_id: 1 cgroup_id_name: container
bpf_trace_printk: tgid: 22345 | cgroup_id: 2 cgroup_id_name: 
bpf_trace_printk: tgid: 22345 | cgroup_id: 3 cgroup_id_name: container
bpf_trace_printk: tgid: 22345 | cgroup_id: 4 cgroup_id_name: container
bpf_trace_printk: tgid: 22345 | cgroup_id: 5 cgroup_id_name: 
bpf_trace_printk: tgid: 22345 | cgroup_id: 6 cgroup_id_name: 
bpf_trace_printk: tgid: 22345 | cgroup_id: 7 cgroup_id_name: 
bpf_trace_printk: tgid: 22345 | cgroup_id: 8 cgroup_id_name: container
bpf_trace_printk: tgid: 22345 | cgroup_id: 9 cgroup_id_name: 
bpf_trace_printk: tgid: 22345 | cgroup_id: 10 cgroup_id_name: container
bpf_trace_printk: tgid: 22345 | cgroup_id: 11 cgroup_id_name: container
bpf_trace_printk: tgid: 22345 | cgroup_id: 12 cgroup_id_name: machine.slice
bpf_trace_printk: tgid: 22345 | cgroup_id: 13 cgroup_id_name: machine.slice
bpf_trace_printk: tgid: 22345 | cgroup_id: 14 cgroup_id_name: 
```

So only few of the are reliable

Reply:
switched to:
```c
  int cgrp_id;
  if (bpf_core_enum_value_exists(enum cgroup_subsys_id, memory_cgrp_id))
    cgrp_id = bpf_core_enum_value(enum cgroup_subsys_id, memory_cgrp_id);
  else
    return -1;
```

resolving this conversation as completed